test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -976,7 +1017,13 @@ void UcxConnection::am_data_recv_callback(void *request, ucs_status_t status,
 
 void UcxConnection::error_callback(void *arg, ucp_ep_h ep, ucs_status_t status)
 {
-    reinterpret_cast<UcxConnection*>(arg)->handle_connection_error(status);
+    UcxConnection *conn = reinterpret_cast<UcxConnection*>(arg);
+
+    if (conn->is_disconnecting()) {
+        return;
+    }
```

Comment:
move this inside conn->handle_connection_error

Reply:
done